src/creosote/cli.py

```diff
@@ -1,19 +1,52 @@
 import argparse
-import glob
+import dataclasses
 import sys
+import typing
+from dataclasses import dataclass, field
 from enum import Enum
+from typing import List, Literal
 
+import toml
 from loguru import logger
 
 from creosote import formatters, parsers, resolvers
 from creosote.__about__ import __version__
 
 
+@dataclass
+class Config:
+    """Structured configuration data.
+
+    It is important that these attributes exactly match
+    the ``dest`` specified in ``add_argument``.
+    """
+
+    format: Literal["default", "no-color", "porcelain"] = "default"  # noqa: A003
+    paths: List[str] = field(default_factory=lambda: ["src"])
+    sections: List[str] = field(default_factory=lambda: ["project.dependencies"])
+    exclude_deps: List[str] = field(default_factory=list)
+    deps_file: str = "pyproject.toml"
+    venvs: List[str] = field(default_factory=lambda: [".venv"])
+    features: List[str] = field(default_factory=list)
+
+
+def load_defaults(src="pyproject.toml") -> Config:
+    """Load pyproject.toml defaults form user config.
+
+    Expects user configuration at ``[tool.creosote]``.
+    """
+    with open(src, "r", encoding="utf-8") as f:
+        project_config = toml.loads(f.read())
```

Comment:
It looks like you always expect this file to be present in the current working directory. Please handle the case where `src` does not exist.
I believe creosote should just continue silently when the file does not exist, as this could be a common use case.

Please also add in debug logging for:
- file exists and is read.
- what gets read from the file's `[tool.creosote]` section.
- file does not exist.

Reply:
added in 2e61193a5e7abd796

I had to refactor the argument parsing a bit so that logging-related configurations are set/parsed as early as possible.

Reply:
whoops! definitely missed that.

* fixed missing file + added a unit test in 0f726c5065464bf7aa00e6.
* Added logging in 2e61193a5e7abd79